### PR TITLE
Add tests for CalculatedAmount

### LINKED ISSUE #115: Bring up code coverage
Code coverage is currently at 69.29%. The aim is to bring this up to 100%.

### PR DESCRIPTION
**Pull request by wltan**
Partially addresses #115 

`CalculatedAmount` should now have 100% line and mutation coverage. Note that `Amount.java` as a whole is still not 100% due to `hashCode` not being tested yet.

### CODE REVIEW

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/CalculatedAmountTest.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    private CalculatedAmount negative = zero.subtract(positive);
+
+    @Test
+    public void zeroValue() {
+        assertEquals(BigDecimal.ZERO, zero.getValue());
+        assertTrue(zero.isNonNegative());
+    }
+
+    @Test
+    public void isNonNegative() {
```
Comment (yongping827) [inline src/test/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/CalculatedAmountTest.java:27]: Maybe add:
```
        assertTrue(zero.isNonNegative());
```
Since it is a boundary value.

**Review by yongping827 — CHANGES_REQUESTED** [review]

File: src/test/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/CalculatedAmountTest.java (hunk 1 above)
Comment (wltan) [reply src/test/java/ay2021s1_cs2103_w16_3/finesse/model/transaction/CalculatedAmountTest.java:27]: I've moved it from the `zeroValue` test, it should make more sense this way.

**Review by ianyong — APPROVED** [review]
LGTM!

**Review by zhaojj2209 — APPROVED** [review]
LGTM!

**Review by yongping827 — APPROVED** [review]
LGTM
